sky/data/mounting_utils.py

```diff
@@ -106,7 +106,8 @@ def get_az_mount_cmd(container_name: str,
     cache_path = _BLOBFUSE_CACHE_DIR.format(
         storage_account_name=storage_account_name,
         container_name=container_name)
-    mount_cmd = (f'AZURE_STORAGE_ACCOUNT={storage_account_name} '
+    mount_cmd = (f'rm -rf {cache_path} && mkdir -p {cache_path} && '
```

Comment:
This can fix 

```
E 12-30 15:48:46 subprocess_utils.py:142] *** blobfuse2: A new version [2.3.2] is available. Consider upgrading to latest version for bug-fixes & new features. *** 
E 12-30 15:48:46 subprocess_utils.py:142] Visit https://aka.ms/blobfuse2warnings#220 to see the list of vulnerabilities associated with your current version [2.2.0] 
E 12-30 15:48:46 subprocess_utils.py:142] Error: failed to initialize new pipeline [config error in file_cache [temp directory not empty]
```

For command `blobfuse2` after sky stop and then start for same cluster.

Failure smoke test:
```
tests/smoke_tests/test_mount_and_storage.py::test_azure_storage_mounts_with_stop --azure
```

Reply:
Leave a comment here for why this fixes the issue.